rabbit-escape-engine/src/rabbitescape/engine/util/Position.java

```diff
@@ -15,4 +15,43 @@ public Position plus( Position p )
     {
         return new Position( this.x + p.x, this.y + p.y );
     }
+
+    /**
+     * Lexical comparison: y then x.
+     */
+    @Override
+    public int compareTo( Position other )
+    {
+        if ( y != other.y )
+        {
+            return y - other.y ;
+        }
+        return x - other.x;
+    }
+
+    @Override
+    public boolean equals( Object o )
+    {
+        if ( null == o )
+        {
+            return false;
+        }
+        if ( !( o instanceof Position ) )
+        {
+            return false;
+        }
+        Position p = (Position)o;
+        return this.x == p.x && this.y == p.y;
+    }
+
+    /**
+     * This may break if we have levels with a dimension greater than 1000
+     * and the class is used for nominal pixels relative to the world origin.
```

Comment:
It won't actually break as such. It's just optimised for x coordinates between 0 and 999. Anyway, good idea to add hash code and equals!